src/helpers/types.ts

```diff
@@ -86,3 +86,24 @@ export type ObjectPaths<T extends Record<string, any>> = Path<T>;
 export type KeysNotOfType<T, U> = {
   [P in keyof T]: T[P] extends U ? never : P;
 }[keyof T];
+
+/**
+ * Returns all keys from objects in the union `T`.
+ */
+type UnionKeys<T> = T extends T ? keyof T : never;
```

Comment:
**question (non-blocking)**: can this be simplified to the following? Is there a case where `T` would not extend `T`?

```suggestion
type UnionKeys<T> = keyof T;
```